server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -133,15 +133,15 @@ public Supplier<RetentionLeases> retentionLeasesSupplier() {
             case "lz4":
             case "best_compression":
             case "zlib":
-            case "zstd":
-            case "zstd_no_dict":
             case "lucene_default":
                 return s;
             default:
+                if (("zstd".equals(s) || "zstd_no_dict".equals(s)) && Codec.availableCodecs().contains("Lucene95CustomCodec")) {
```

Comment:
Where `Lucene95CustomCodec` is coming from?

Reply:
```Lucene95CustomCodec``` is the actual codec name where as ```zstd``` and ```zstd_no_dict``` are the modes of the codec.

Reply:
@reta That's the [key](https://github.com/apache/lucene/blob/d1c353116157d0375de9d673ae5e9c90524ffe2f/lucene/core/src/java/org/apache/lucene/codecs/FilterCodec.java#L55) for registering OpenSearch's [Lucene95CustomCodec](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/codec/customcodecs/Lucene95CustomCodec.java) through Lucene's NamedSPILoader. The OpenSearch custom codec (from Elasticsearch) enables users to change compression levels.